internal/pkg/agent/application/dispatcher/dispatcher.go

```diff
@@ -117,19 +128,41 @@ func (ad *ActionDispatcher) Dispatch(ctx context.Context, acker acker.Acker, act
 		strings.Join(detectTypes(actions), ", "),
 	)
 
+	var mErr error
 	for _, action := range actions {
-		if err := ctx.Err(); err != nil {
-			return err
+		if err = ctx.Err(); err != nil {
+			ad.errCh <- err
+			return
 		}
 
 		if err := ad.dispatchAction(ctx, action, acker); err != nil {
+			var e errors.Error
+			if errors.As(err, &e) && e.Type() == errors.TypeRetryableAction {
+				rAction, ok := action.(fleetapi.RetryableAction)
+				if !ok {
+					// This line should only occur if an action hander returns a retryable error
+					// but the action defined in fleetapi/action.go does not implement the required methods.
+					ad.log.Warnf("action ID %s is not a retryable action, handler retuned error: %v", action.ID(), err)
+					mErr = multierror.Append(mErr, err)
+					continue
+				}
+				rAction.SetError(e) // set the retryable action error to what the dispatcher returned
+				ad.scheduleRetry(ctx, rAction, acker)
+				continue
+			}
 			ad.log.Debugf("Failed to dispatch action '%+v', error: %+v", action, err)
-			return err
+			mErr = multierror.Append(mErr, err)
```

Comment:
Why use a multierror here? Why not just sent the error over the errCh, each time one occurs?

Reply:
👍 this was done before i made the error channel for the dispatcher. have removed